UnitySDK/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -551,18 +556,40 @@ void EnvironmentStep()
 
             AgentSetStatus(m_StepCount);
 
-            AgentResetIfDone();
+            using (m_Timer.Scoped("AgentResetIfDone"))
+            {
+                AgentResetIfDone();
+            }
 
-            AgentSendState();
+            using (m_Timer.Scoped("AgentSendState"))
+            {
+                AgentSendState();
+            }
 
-            BrainDecideAction();
+            using (m_Timer.Scoped("BrainDecideAction"))
+            {
+                BrainDecideAction();
+            }
 
-            AcademyStep();
+            using (m_Timer.Scoped("AcademyStep"))
+            {
+                AcademyStep();
+            }
 
-            AgentAct();
+            using (m_Timer.Scoped("AgentAct"))
+            {
+                AgentAct();
+            }
 
             m_StepCount += 1;
             m_TotalStepCount += 1;
+
+            if (m_TotalStepCount % 100 == 0)
+            {
+                // TODO better place to hook this in?
+                // TODO append environment ID so multiple envs don't overwrite each other
```

Comment:
Talked to @harper-u3d about this; might pass the worker ID as an env variable.

Reply:
makes sense.